File: gst-caps-to-rfc6381.py
#!/usr/bin/env python3
import gi
import sys
from argparse import ArgumentParser
gi.require_version('Gst', '1.0')
gi.require_version('GstPbutils', '1.0')
gi.require_version('GLib', '2.0')
from gi.repository import GLib, Gst, GstPbutils # type: ignore

def main():
    parser = ArgumentParser()
    parser.add_argument("caps_string")

    args = parser.parse_args()

    Gst.init(sys.argv)
    caps = Gst.Caps.from_string(args.caps_string)
    if not caps:
        print("Failed to create caps.", file=sys.stderr)
        raise SystemExit(-1)
    mime_codec = GstPbutils.codec_utils_caps_get_mime_codec(caps)
    if not mime_codec:
        print("Could not create a MIME codec string.", file=sys.stderr)
        raise SystemExit(-1)
    print(mime_codec)

if __name__ == "__main__":
    main()

File: gst-caps-to-rfc6381.c
#include <gst/gst.h>
#include <gst/pbutils/codec-utils.h>
#include <glib.h>

int
main (int   argc,
      char *argv[])
{
  gst_init (&argc, &argv);

  if (argc != 2) {
    g_printerr ("Usage: %s <caps-string>\n", argv[0]);
    return -1;
  }

  GstCaps* caps = gst_caps_from_string (argv[1]);
  if (!caps) {
    g_printerr("Failed to create caps.\n");
    return -1;
  }
  gchar* mime_codec = gst_codec_utils_caps_get_mime_codec (caps);
  if (!mime_codec) {
    g_printerr ("Could not create a MIME codec string.\n");
    return -1;
  }

  g_print ("%s\n", mime_codec);

  g_free (mime_codec);
  gst_caps_unref (caps);
  return 0;
}
